packages/frontpage/app/(app)/post/new/_action.ts

```diff
@@ -22,6 +22,13 @@ export async function newPostAction(_prevState: unknown, formData: FormData) {
     return { error: "Title too long" };
   }
 
+  const allowedProtocols = ["http", "https", "at"];
+  const protocolRegex = new RegExp(`^(${allowedProtocols.join("|")}):\/\/`);
+
+  if (!protocolRegex.test(url)) {
+    url = `https://${url}`;
```

Comment:
I'm not sure it's a good idea to transform http URLs (that we accept in the above regex) to https. A link may not respond to http.

It's worth seeing what other apps do here (eg. reddit, hacker news, lemmy) to see if there's consensus.

Reply:
Hmm, it should not transform `http` to `https`, it should just be looking to see if the url starts with one of the allowed protocols.

I did start off with a simpler implementation that I might fall back too.